core/tee/tee_ree_fs.c

```diff
@@ -498,12 +498,19 @@ static TEE_Result open_dirh(struct tee_fs_dirfile_dirh **dirh)
 		return res;
 
 	res = tee_fs_dirfile_open(false, hashp, &ree_dirf_ops, dirh);
+
 	if (res == TEE_ERROR_ITEM_NOT_FOUND) {
-		if (IS_ENABLED(CFG_REE_FS_ALLOW_RESET) && hashp) {
-			DMSG("Clear REE FS hash in RPMB");
-			res = rpmb_fs_ops.truncate(ree_fs_rpmb_fh, 0);
-			if (res) {
-				DMSG("Clear REE FS hash failed: %#"PRIx32, res);
+		if (hashp) {
+			if (IS_ENABLED(CFG_REE_FS_ALLOW_RESET)) {
+				DMSG("Clear REE FS hash in RPMB");
+				res = rpmb_fs_ops.truncate(ree_fs_rpmb_fh, 0);
+				if (res) {
+					DMSG("Can't clear hash: %#"PRIx32, res);
+					res = TEE_ERROR_SECURITY;
+					goto out;
+				}
+			} else {
+				DMSG("Secure storage corruption detected");
```

Comment:
- Should this one be `EMSG()` perhaps?
- I suggest mentioning `dirf.db` to help troubleshooting, because that file name is hidden quite far from here. How about "Secure storage corruption detected (no dirf.db)"? (ignoring the long line >80 characters)

Reply:
I used `DMSG()` as around line 427 (`DMSG("Secure storage corruption detected");`)

Maybe `DMSG("Expected dirf.db file not found");` ? (fits in 80chars)

Reply:
> I used `DMSG()` as around line 427 (`DMSG("Secure storage corruption detected");`)

OK, let's be consistent then.

> Maybe `DMSG("Expected dirf.db file not found");` ? (fits in 80chars)

Good, you can even drop "Expected".